### PR TITLE
Use `SyscallRequestWrapper` in `TestExecutionSyscallHandler`

### LINKED ISSUE #837: Investigate using `SyscallRequestWrapper::<Request>::read` in `TestExecutionSyscallHandler`
We currently do a raw read using syscall handler:
https://github.com/foundry-rs/starknet-foundry/blob/07afb7ea51809d18477680cc3e0ad523318911a4/crates/forge/src/test_execution_syscall_handler.rs#L725

Maybe we could use existing abstraction instead as it is used here 
https://github.com/foundry-rs/starknet-foundry/blob/07afb7ea51809d18477680cc3e0ad523318911a4/crates/cheatnet/src/execution/syscalls.rs#L404

### PR DESCRIPTION
**Pull request by war-in**
Closes #837

## Introduced changes

- `cheatable_syscall_handler.syscall_handler.system_ptr` is used instead of `system_ptr`
- `SyscallRequestWrapper` is used instead of `MemBuffer`

## Checklist

- [x] Linked relevant issue
~- [ ] Updated relevant documentation~
~- [ ] Added relevant tests~
- [x] Performed self-review of the code
~- [ ] Added changes to `CHANGELOG.md`~

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]
LGTM

File: crates/forge/src/test_execution_syscall_handler.rs [hunk 1]
```diff
@@ -635,25 +637,24 @@ struct ScarbStarknetContractArtifact {
...
 fn execute_syscall(
-    system: &ResOperand,
     vm: &mut VirtualMachine,
     exec_scopes: &mut ExecutionScopes,
     hint_data: &Box<dyn Any>,
     constants: &HashMap<String, Felt252>,
     cheatable_syscall_handler: &mut CheatableSyscallHandler,
 ) -> Result<(), HintError> {
-    let (cell, offset) = extract_buffer(system);
-    let system_ptr = get_ptr(vm, cell, &offset)?;
```
Comment (MaksymilianDemitraszek) [inline crates/forge/src/test_execution_syscall_handler.rs:646]: Can we execute `validate_syscall_ptr` here? to make sure `system_ptr` and `cheatable_syscall_handler.syscall_handler.syscall_ptr` are equal?

File: crates/forge/src/test_execution_syscall_handler.rs [hunk 2]
```diff
@@ -689,58 +684,71 @@ struct CallContractArgs {
...
     vm: &mut VirtualMachine,
     call_args: &CallContractArgs,
     call_output: CallContractOutput,
 ) -> Result<(), HintError> {
-    let mut buffer = MemBuffer::new(vm, system_ptr);
-
-    let (result, exit_code) = match call_output.result {
-        CallContractResult::Success { ret_data, .. } => (ret_data, 0),
+    let response_wrapper: SyscallResponseWrapper<SingleSegmentResponse> = match call_output.result {
+        CallContractResult::Success { ret_data, .. } => {
```
[the remark this answers is not in the document]
Comment (war-in) [reply crates/forge/src/test_execution_syscall_handler.rs:725]: Will create a task for this

[the remark this answers is not in the document]
Comment (war-in) [reply crates/forge/src/test_execution_syscall_handler.rs:725]: #927

**Pull request merged**